hibernate-core/src/main/java/org/hibernate/jpa/event/internal/CallbackDefinitionResolver.java

```diff
@@ -282,7 +283,10 @@ public static void resolveLifecycleCallbacks(
 		//       code was added by HHH-12326
 		collector.addSecondPass( persistentClasses -> {
 			for ( Property property : persistentClass.getDeclaredProperties() ) {
-				if ( property.isComposite() ) {
+			if ( property.isComposite()
+				// this is to avoid issuing an embeddableLifecycleCallback warning in case the embedded property represents an IdClass (see HHH-19671)
+				// alternative way: collector.getClassDetailsRegistry().getClassDetails( property.getType().getReturnedClass().getName() ).hasDirectAnnotationUsage( IdClass.class )
+				&& !NavigablePath.IDENTIFIER_MAPPER_PROPERTY.equals(property.getName()) ) {
```

Comment:
```suggestion
				if ( property.getValue() instanceof Component component
					// embedded components don't have their own class, so no need to check callbacks
					&& !component.isEmbedded() ) {
```

Reply:
Your previous approach only considered the "embedded component" for non-aggregated identifiers, but through XML mappings or mapping `@OneToMany(mappedBy = "...")` with `@ManyToOne @JoinColumns` that are not the PK, I believe that there are other possible "embedded component" cases that could happen. Happy to help if you want to write test cases for that.

Reply:
Not sure if it's worth it? There isn't a specific test case for the IdClass warning either; it did occur in a test case but the actual warning wasn't being tested, and also it's my understanding (correct me if wrong) that Gavin doesn't favor tests that do this (check log messages).